File: Arrays & Hashing/Valid Anagram(optimized).cs
public class Solution {
       public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length)
            return false;
        
        //char[] arrays can be sorted. Neat!
        char[] source = s.ToCharArray(), target = t.ToCharArray();
        
        Array.Sort(source);
        Array.Sort(target);
        
        for (int i = 0; i <= source.Length - 1; i++)
            if(source[i] != target[i])
                return false;
        
        return true;
    }
}

File: Arrays & Hashing/Valid Anagram(optimized).ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length) return false;

    const charCount = {};

    for (let i = 0; i < s.length; i++) {
        charCount.hasOwnProperty(s[i]) ? charCount[s[i]]++ : (charCount[s[i]] = 1);
        charCount.hasOwnProperty(t[i]) ? charCount[t[i]]-- : (charCount[t[i]] = -1);
    }

    for (const key in charCount) {
        if (charCount[key] !== 0) return false;
    }

    return true;
}
